core/src/main/java/com/orientechnologies/orient/core/sql/operator/OQueryOperatorMatches.java

```diff
@@ -66,6 +69,21 @@ private boolean matches(String iValue, String iRegex, OCommandContext iContext)
 			p = Pattern.compile(iRegex);
 			iContext.setVariable(key, p);
 		}
-		return p.matcher(iValue).matches();
+		Matcher m = p.matcher(iValue);
+		if (m.matches()) {
+			List<String> iMatches = new ArrayList<String>();
+			List<Integer> iGroups = new ArrayList<Integer>();
+			for(int i = 1; i <= m.groupCount(); i++) {
+				String match = m.group(i);
+				if (match != null) {
+					iMatches.add(match);
+					iGroups.add(i-1);
+				}
+			}
+			iContext.setVariable("matches", iMatches);

```

Comment:
also while we're at it, we should consider to refactor all of the available context variable names into a constants file so that it becomes easier to document what context variables are available. it should probably be a new issue though.